programs/stake/benches/stake_iai.rs

```diff
@@ -0,0 +1,770 @@
+                invoke_context.mock_set_feature_set(Arc::clone(&self.feature_set));
+            },
+            |_invoke_context| {},
+        );
+        // update stake account
+        self.transaction_accounts[0] = (self.stake_address, accounts[0].clone());
+
+        withdraw_authority_address
+    }
+
+    fn delegate_stake(&mut self) {
+        let vote_address = Pubkey::new_unique();
+
+        let instruction =
+            instruction::delegate_stake(&self.stake_address, &self.stake_address, &vote_address);
+
+        let transaction_accounts = vec![
+            (self.stake_address, self.stake_account.clone()),
+            (
+                vote_address,
+                vote_state::create_account(&vote_address, &Pubkey::new_unique(), 0, 100),
+            ),
+            (
+                clock::id(),
+                create_account_shared_data_for_test(&Clock::default()),
+            ),
+            (
+                stake_history::id(),
+                create_account_shared_data_for_test(&StakeHistory::default()),
+            ),
+        ];
+
+        let accounts = mock_process_instruction(
+            &solana_stake_program::id(),
+            Vec::new(),
+            &instruction.data,
+            transaction_accounts,
+            instruction.accounts.clone(),
+            Ok(()),
+            stake_instruction::Entrypoint::vm,
+            |invoke_context| {
+                invoke_context.mock_set_feature_set(Arc::clone(&self.feature_set));
+            },
+            |_invoke_context| {},
+        );
+        self.stake_account = accounts[0].clone();
+        self.stake_account.set_lamports(u64::MAX - 100); // Hack: replenish stake account, but not to hit
+                                                         // MAX.
+                                                         // update stake account
+        self.transaction_accounts[0] = (self.stake_address, self.stake_account.clone());
+    }
+
+    fn run(&self) {
+        mock_process_instruction(
+            &solana_stake_program::id(),
+            Vec::new(),
+            &self.instruction_data,
+            self.transaction_accounts.clone(),
+            self.instruction_accounts.clone(),
+            Ok(()), //expected_result,
+            stake_instruction::Entrypoint::vm,
+            |invoke_context| {
+                invoke_context.mock_set_feature_set(Arc::clone(&self.feature_set));
+            },
+            |_invoke_context| {},
+        );
+    }
+}
+
+fn bench_setup_initialize() -> TestSetup {
+    let mut test_setup = TestSetup::new();
+    test_setup.add_account(
+        solana_sdk::sysvar::rent::id(),
+        create_account_shared_data_for_test(&Rent::default()),
+    );
+
+    test_setup.instruction_data = serialize(&StakeInstruction::Initialize(
+        Authorized::auto(&test_setup.stake_address),
+        Lockup::default(),
+    ))
+    .unwrap();
+
+    test_setup
+}
+
+fn bench_initialize() {
+    bench_setup_initialize().run();
```

Comment:
still too many steps in `run()` need to be extracted into `bench_setup_initialize()`